src/games/TicTacToe.ts

```diff
@@ -0,0 +1,248 @@
+            }
+
+            const oplayer = game.channel.members.find(m => m.id === game.Oplayer);
+            const xplayer = game.channel.members.find(m => m.id === game.Xplayer);
+
+            const highP = high ? oplayer.user.username : xplayer.user.username;
+            const lowP = high ? xplayer.user.username : oplayer.user.username;
+            const highScore = this.scores[game.scoreKey].highWins;
+            const lowScore = this.scores[game.scoreKey].lowWins;
+            const drawScore = this.scores[game.scoreKey].draws;
+
+            game.channel.send("The game has ended!\nScores are now: " + highP + "(" + highScore + ") - " + drawScore + " - " + lowP + "(" + lowScore + ")");
+            game.channel.overwritePermissions(oplayer, { SEND_MESSAGES: false });
+            game.channel.overwritePermissions(xplayer, { SEND_MESSAGES: false });
+            this.games = this.games.filter(g => g !== game);
+            return;
+        }
+
+        this.doBottyMove(game);
+    }
+
+    /**
+     * returns "-1" for a draw
+     * returns undefined for unfinished
+     * otherwise the playerid of the winner
+     */
+    private getGameWinner(game: TicTacToeGame) {
+
+        let player = TicTacToeType.X;
+        if ((game.board[1] === player && game.board[2] === player && game.board[3] === player) ||
+            (game.board[4] === player && game.board[5] === player && game.board[6] === player) ||
+            (game.board[7] === player && game.board[8] === player && game.board[9] === player) ||
+            (game.board[1] === player && game.board[4] === player && game.board[7] === player) ||
+            (game.board[2] === player && game.board[5] === player && game.board[8] === player) ||
+            (game.board[3] === player && game.board[6] === player && game.board[9] === player) ||
+            (game.board[1] === player && game.board[5] === player && game.board[9] === player) ||
+            (game.board[3] === player && game.board[5] === player && game.board[7] === player)) {
+            return game.Xplayer;
+        }
+        player = TicTacToeType.O;
+        if ((game.board[1] === player && game.board[2] === player && game.board[3] === player) ||
+            (game.board[4] === player && game.board[5] === player && game.board[6] === player) ||
+            (game.board[7] === player && game.board[8] === player && game.board[9] === player) ||
+            (game.board[1] === player && game.board[4] === player && game.board[7] === player) ||
+            (game.board[2] === player && game.board[5] === player && game.board[8] === player) ||
+            (game.board[3] === player && game.board[6] === player && game.board[9] === player) ||
+            (game.board[1] === player && game.board[5] === player && game.board[9] === player) ||
+            (game.board[3] === player && game.board[5] === player && game.board[7] === player)) {
+            return game.Oplayer;
+        }
+
+        for (let i = 1; i < 10; i++) {
```

Comment:
if (game.board.some(x=>!x)) return;

return is the same as return undefined

Reply:
board is a "map" type, not an array, so .some() is not an option